pallets/automation-time/src/migrations.rs

```diff
@@ -1,72 +1,327 @@
+				storage_key_iter::<T::Hash, Task<T>, Twox64Concat>(pallet_prefix, old_tasks_prefix)
+					.drain()
+					.collect::<btree_map::BTreeMap<T::Hash, Task<T>>>();
+			old_tasks.iter().for_each(|(task_id, task)| {
+				AccountTasks::<T>::insert(task.owner_id.clone(), task_id, task.clone());
+			});
+			let task_migrated = old_tasks.len() as u64;
+
+			// Move all tasks from ScheduledTasks to ScheduledTasksV2
+			let mut times_migrated = 0u64;
+			let old_scheduled_prefix: &[u8] = b"ScheduledTasks";
+			storage_key_iter::<u64, BoundedVec<T::Hash, T::MaxTasksPerSlot>, Twox64Concat>(
+				pallet_prefix,
+				old_scheduled_prefix,
+			)
+			.drain()
+			.for_each(|(time, task_ids)| {
+				let new_task_ids = task_ids
+					.into_iter()
+					.filter_map(|task_id| {
+						if let Some(task) = old_tasks.get(&task_id) {
+							Some((task.owner_id.clone(), task_id))
+						} else {
+							log::debug!(target: "automation-time", "Unable to get task with id {:?}", task_id);
+							None
+						}
+					})
+					.collect::<Vec<_>>();
+
+				let account_task_ids: BoundedVec<AccountTaskId<T>, T::MaxTasksPerSlot> =
+					new_task_ids.try_into().unwrap();
+				ScheduledTasksV2::<T>::insert(time, account_task_ids);
+				times_migrated += 1;
+			});
+
+			// Move all tasks from TaskQueue to TaskQueueV2
+			let old_task_queue_prefix: &[u8] = b"TaskQueue";
+			let new_task_ids =
+				take_storage_value::<Vec<TaskId<T>>>(pallet_prefix, old_task_queue_prefix, &[])
+					.unwrap_or(vec![])
+					.into_iter()
+					.filter_map(|task_id| {
+						if let Some(task) = old_tasks.get(&task_id) {
+							Some((task.owner_id.clone(), task_id))
+						} else {
+							log::debug!(target: "automation-time", "Unable to get task with id {:?}", task_id);
+							None
+						}
+					})
+					.collect::<Vec<_>>();
+			TaskQueueV2::<T>::put(new_task_ids);
+
+			// Move all tasks from MissedQueue to MissedQueueV2, and convert from MissedTask to MissedTaskV2
+			let old_missed_queue_prefix: &[u8] = b"MissedQueue";
+			let new_missed_tasks = take_storage_value::<Vec<MissedTask<T>>>(
+				pallet_prefix,
+				old_missed_queue_prefix,
+				&[],
+			)
+			.unwrap_or(vec![])
+			.into_iter()
+			.filter_map(|missed_task| {
+				if let Some(task) = old_tasks.get(&missed_task.task_id) {
+					Some(MissedTaskV2::<T>::create_missed_task(
+						task.owner_id.clone(),
+						missed_task.task_id,
+						missed_task.execution_time,
+					))
+				} else {
+					log::debug!(target: "automation-time", "Unable to get task with id {:?}", missed_task.task_id);
+					None
+				}
+			})
+			.collect::<Vec<_>>();
+			MissedQueueV2::<T>::put(new_missed_tasks);
+
+			// Set new storage version and return weight
+			StorageVersion::new(3).put::<Pallet<T>>();
+
+			// For each task there is
+			// 1 read to get it into memory from Tasks
+			// 1 write to remove it from Tasks
+			// 1 write to add it to AccountTasks
+			let weight = T::DbWeight::get().reads_writes(task_migrated + 1, task_migrated * 2);
```

Comment:
let mut necessary? or is it better to just redefine the variable each time?

Reply:
I like the redefine for this sort of thing

Reply:
just for my information, what's the diff? i.e. when should i redefine and when not?

Reply:
it's more a style thing. I don't know a hard and fast rule yet